src/mlpack/methods/kde/kde_rules_impl.hpp

```diff
@@ -26,23 +29,35 @@ KDERules<MetricType, KernelType, TreeType>::KDERules(
     arma::vec& densities,
     const double relError,
     const double absError,
+    const double mcProb,
+    const size_t initialSampleSize,
+    const double mcAccessCoef,
+    const double mcBreakCoef,
     MetricType& metric,
     KernelType& kernel,
+    const bool monteCarlo,
     const bool sameSet) :
     referenceSet(referenceSet),
     querySet(querySet),
     densities(densities),
     absError(absError),
     relError(relError),
+    mcBeta(1 - mcProb),
+    initialSampleSize(initialSampleSize),
+    mcAccessCoef(mcAccessCoef),
+    mcBreakCoef(mcBreakCoef),
     metric(metric),
     kernel(kernel),
+    monteCarlo(monteCarlo),
     sameSet(sameSet),
     lastQueryIndex(querySet.n_cols),
     lastReferenceIndex(referenceSet.n_cols),
     baseCases(0),
     scores(0)
 {
-  // Nothing to do.
+  // Initialize accumMCAlpha only if Monte Carlo estimations are available.
+  if (monteCarlo && kernelIsGaussian)
+    accumMCAlpha = std::move(arma::vec(querySet.n_cols, arma::fill::zeros));
```

Comment:
I think the `std::move()` is superfluous here, and I think return value optimization (RVO) will handle the optimization here (but not if `std::move()` is used).